static/app/views/explore/components/schemaHintsDrawer.tsx

```diff
@@ -59,36 +85,63 @@ function SchemaHintsDrawer({hints}: SchemaHintsDrawerProps) {
     [exploreQuery, setExploreQuery]
   );
 
+  const noAttributesMessage = (
+    <NoAttributesMessage>
+      <p>{t('No attributes found.')}</p>
+    </NoAttributesMessage>
+  );
   return (
     <Fragment>
       <DrawerHeader hideBar />
       <DrawerBody>
         <HeaderContainer>
           <SchemaHintsHeader>{t('Filter Attributes')}</SchemaHintsHeader>
-          <IconSearch size="md" />
+          <InputGroup
+            css={css`
+              @media (max-width: ${theme.breakpoints.medium}) {
+                max-width: 175px;
+              }
+            `}
+          >
+            <SearchInput
+              size="sm"
+              value={searchQuery}
+              onChange={e => setSearchQuery(e.target.value)}
+              aria-label={t('Search attributes')}
+            />
+            <InputGroup.TrailingItems disablePointerEvents>
+              <IconSearch size="md" />
+            </InputGroup.TrailingItems>
+          </InputGroup>
         </HeaderContainer>
         <StyledMultipleCheckbox name={t('Filter keys')} value={selectedFilterKeys}>
-          {sortedHints.map(hint => {
-            const hintFieldDefinition = getFieldDefinition(hint.key, 'span', hint.kind);
-            const hintType =
-              hintFieldDefinition?.valueType === FieldValueType.BOOLEAN
-                ? t('boolean')
-                : hint.kind === FieldKind.MEASUREMENT
-                  ? t('number')
-                  : t('string');
-            return (
-              <StyledMultipleCheckboxItem
-                key={hint.key}
-                value={hint.key}
-                onChange={() => handleCheckboxChange(hint)}
-              >
-                <CheckboxLabelContainer>
-                  <CheckboxLabel>{prettifyTagKey(hint.key)}</CheckboxLabel>
-                  <Badge>{hintType}</Badge>
-                </CheckboxLabelContainer>
-              </StyledMultipleCheckboxItem>
-            );
-          })}
+          {sortedAndFilteredHints.length === 0
+            ? noAttributesMessage
+            : sortedAndFilteredHints.map(hint => {
+                const hintFieldDefinition = getFieldDefinition(
+                  hint.key,
+                  'span',
+                  hint.kind
+                );
+                const hintType =
+                  hintFieldDefinition?.valueType === FieldValueType.BOOLEAN
+                    ? t('boolean')
+                    : hint.kind === FieldKind.MEASUREMENT
+                      ? t('number')
+                      : t('string');
```

Comment:
nit: I wonder if we could memoize field definitions and hint types when we build the initial list so we don't have to recompute it on every filter change